cylc/flow/graph_parser.py

```diff
@@ -705,6 +708,11 @@ def _families_all_to_all(
             else:
                 n_info += [(name, offset, trig)]
 
+            # Check output optionality consistency for left side nodes.
+            # (We don't infer output optionality from right side nodes).
+            # Note suicide is False by definition for left side nodes.
+            self._set_output_opt(name, trig, opt, False, fam_member)
```

Comment:
Is this not called as part of `_compute_triggers`?

Reply:
Hmm, good spot. I don't think that will have any deleterious effects, but best not to do it ... checking ...

Reply:
> Is this not called as part of `_compute_triggers`?

Yes, but it is only called for right-side nodes there. 

On master, that was it, because optionality was only set and checked for right-side nodes. (Because all nodes appear on the right at some point, even lone and first-in-chain nodes appear as `None => foo` pairs). 

On this branch optionality is set and checked definitively for left-side nodes and only for right-side nodes with explicit outputs (or explicit `?` short for `:succeeded?`)

```
foo:out => bar  # foo:out required; says nothing about bar:succeeded
foo:out => bar?  # foo:out required; bar:succeeded optional
```